[feat] 지역(동네) 선택 기능

pie2457 opened the pull request:
## Issues
- #41 

## What is this PR? 👓
- 지역(동네) 선택 기능에 대한 PR입니다.

## Key changes 🔑
- 지역(동네) 선택 기능
- 지역(동네) 선택 기능 테스트

pie2457 committed:
feat #41 : 지역(동네) 선택 기능

pie2457 committed:
test #41 : 지역(동네) 선택 기능 테스트

pie2457 committed:
style #41 : 변수명 toEntity -> of로 수정

pie2457 committed:
style #41 : AllArgsConstructor accessLevel 설정

23Yong commented on src/main/java/pie/tomato/tomatomarket/application/memberTown/MemberTownService.java:
```diff
@@ -61,4 +62,26 @@ private Member getMember(Principal principal) {
+		MemberTown selectMemberTown = getMemberTown(selectMemberTownRequest, memberTowns);
+		selectMemberTown.changeIsSelectedTrue();
+
+		for (MemberTown memberTown : memberTowns) {
+			if (!memberTown.isSameRegionId(selectMemberTown.getId())) {
+				memberTown.changeIsSelectedFalse();
+			}
+		}
```
이 부분 stream을 써서 바꿀 수 있을 것 같은데요?!

pie2457 replied:
음..!!! 한번 해보겠습니다 !!!

23Yong commented on src/test/java/pie/tomato/tomatomarket/application/MemberTownServiceTest.java:
```diff
@@ -90,6 +91,28 @@ void addMemberTown_ver3() {
+		// then
+		MemberTown SelectMemberTown = supportRepository.findById(1L, MemberTown.class);
+		MemberTown nonSelectMemberTown = supportRepository.findById(2L, MemberTown.class);
+		assertThat(SelectMemberTown.isSelected()).isTrue();
+		assertThat(nonSelectMemberTown.isSelected()).isFalse();
```
assertAll이나 softAssertion으로 묶으면 좋을 것 같네요~

pie2457 replied:
앗 바꾼다는걸 깜빡했네요 감사합니당 ㅎㅎ

23Yong reviewed (approved):
수고하셨어여~

pie2457 committed:
refactor #41 : 리뷰사항 반영

- MemberTownService : for문을 stream으로 변경
- MemberTownServiceTest : assertAll로 묶어서 테스트
